src/osyris/spatial/spatial.py

```diff
@@ -0,0 +1,89 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/nvaytet/osyris)
+
+import numpy as np
+from ..core import Array
+from ..config import units
+
+
+def rotation_matrix(vec, angle):
+    """
+    Returns 3D rotation matrix of angle 'angle' around rotation vector 'vec'.
+    """
+    if isinstance(vec, list):
+        vec = np.array(vec)
+    vec = vec / np.linalg.norm(vec)
+    R = np.cos(angle) * np.identity(3) + (np.sin(angle)) * np.cross(
+        vec,
+        np.identity(vec.shape[0]) * -1) + (1 - np.cos(angle)) * (np.outer(vec, vec))
+    return R
+
+
+def spherical_to_cartesian(r, theta, phi):
```

Comment:
I appreciate you wanting to make a start on this earlier rather than later, but as I've said before I think it's worth taking our time to think properly about how we want to implement this.

We have to ask ourselves what is the goal here.
If we want to be able to e.g. easily compute a radius, and make a radial profile, then these functions would definitely be useful.

If we wish to make maps that depend on theta and phi (which was your original motivation), we also need to consider how this would then be used to make the maps. Maybe this approach does work, I am not sure, I have not thought enough about it.

Another approach could have been to give the Array additional properties, such as `.r`, `.theta`, and `.phi`, which would compute the radius and angles on-the-fly. I am thinking that the coordinates are inherently cartesian, because they are as such inside Ramses. So they should be loaded as such, and am therefore not sure that converting from spherical back to cartesian would be something that is ever needed?

In addition, as you pointed out when you first suggested this idea, velocity or B field vectors could also have radial, theta and phi components (and those calculations would be a little different from that of the positions), which would certainly be useful.

This last point, combined with the slicing inconsistency of vectors in the other PR, is making me wonder whether we want a slightly different class for vector arrays (that would inherit from Array).

Reply:
> Another approach could have been to give the Array additional properties, such as .r, .theta, and .phi, which would compute the radius and angles on-the-fly.

> In addition, as you pointed out when you first suggested this idea, velocity or B field vectors could also have radial, theta and phi components (and those calculations would be a little different from that of the positions), which would certainly be useful.

Maybe we can group these into `.get_spherical_components()` & `.get_cylindrical_components()` methods? These shouldn't be automatically computed by osyris upon loading the dataset because otherwise it comes with an extra RAM burden, but they can provide a quick and easy way to gather all components.

> This last point, combined with the slicing inconsistency of vectors in the other PR, is making me wonder whether we want a slightly different class for vector arrays (that would inherit from Array).

Instead of having a different class for vectors, I would rather have the type be an attribute of the Array class (`eg. a .vector boolean attribute`).